spring-rabbit/src/main/java/org/springframework/amqp/rabbit/annotation/RabbitListenerAnnotationBeanPostProcessor.java

```diff
@@ -276,37 +266,28 @@ public Object postProcessBeforeInitialization(Object bean, String beanName) thro
 	@Override
 	public Object postProcessAfterInitialization(final Object bean, final String beanName) throws BeansException {
 		Class<?> targetClass = AopUtils.getTargetClass(bean);
-		if (this.methodCache.get(targetClass) == null) {
-			findMethods(targetClass);
-		}
-		List<RabbitListener> classLevelListeners = this.classAnnotationCache.get(targetClass);
-		List<Method> methods = this.methodCache.get(targetClass);
-		List<Method> multiMethods = this.multiMethodCache.get(targetClass);
-		for (Method method : methods) {
-			List<RabbitListener> listenerAnnotations = this.annotationCache.get(method);
-			if (listenerAnnotations != null) {
-				for (RabbitListener rabbitListener : listenerAnnotations) {
-					processAmqpListener(rabbitListener, method, bean, beanName);
-				}
+		final TypeMetadata metadata = this.typeCache.computeIfAbsent(targetClass, TypeMetadataHolder::new).get();
```

Comment:
We can't use `computeIfAbsent()` if we are going to back port it.
Or... we will need one more backport PR.